frontend/src/scenes/project-homepage/projectHomepageLogic.tsx

```diff
@@ -0,0 +1,35 @@
+import { kea } from 'kea'
+
+import { projectHomepageLogicType } from './projectHomepageLogicType'
+import { teamLogic } from 'scenes/teamLogic'
+import { dashboardLogic } from 'scenes/dashboard/dashboardLogic'
+import { DashboardPlacement } from '~/types'
+
+export const projectHomepageLogic = kea<projectHomepageLogicType>({
+    path: ['scenes', 'project-homepage', 'projectHomepageLogic'],
+
+    selectors: {
+        primaryDashboardId: [() => [teamLogic.selectors.currentTeam], (currentTeam) => currentTeam?.primary_dashboard],
+        dashboardLogic: [
+            (s) => [s.primaryDashboardId],
+            (primaryDashboardId): ReturnType<typeof dashboardLogic.build> | null =>
+                dashboardLogic.build(
+                    { id: primaryDashboardId ?? undefined, placement: DashboardPlacement.ProjectHomepage },
+                    false
+                ),
```

Comment:
Should the logic be set to null if the `primaryDashboardId` isn't set? 

I don't think it's an issue right now because the dashboard isn't shown in the "no primaryDashboardId" case, and at the same time, I don't see how the `useValues` would work for a potentially null logic. Just a thought

Reply:
Basically, `useValues(logic)` requires a logic to be deterministic. React hooks must execute in order, and having or not having a logic makes a big difference, as each fetched `value` runs a `useSelector` hook. If there's no logic, we don't have anything to give useSelector, and we crash in one way or another.

Reply:
This was the lesser of two evils basically.